### PR TITLE
Git Server

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: git-server/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+async fn git_handler(
+    ctx: Context,
+    method: Method,
+    headers: HeaderMap,
+    body: impl Buf,
+    remote: Option<net::SocketAddr>,
+    path: warp::filters::path::Tail,
+    query: String,
+) -> Result<impl Reply, Rejection> {
+    let remote = remote.expect("there is always a remote for HTTP connections");
```
Comment (xla) [inline git-server/src/lib.rs:73]: Why make it an `Option` then?

**Review by xla — APPROVED** [review]
Would probably name it `git-http-server` to communicate the protocol. </bikeshed>

🕴 ⭐️ ⚠️ 👰

File: git-server/src/lib.rs (hunk 1 above)
Comment (cloudhead) [reply git-server/src/lib.rs:96]: It's `warp` that gives me an `Option` :<

Claimed reason being: the transport may not have a remote addr :shrug:

**Comment by cloudhead** [comment]
> Would probably name it `git-http-server` to communicate the protocol.

That was actually the name of it for a good 15 minutes. Keeping it a bit more general for now as I may include SSH-related stuff in there too.

File: git-server/src/lib.rs (hunk 1 above)
Comment (cloudhead) [reply git-server/src/lib.rs:96]: I'm guessing it's because they support like, unix domain socket as a transport? idk..

Comment (cloudhead) [reply git-server/src/lib.rs:96]: Proof enough that warp isn't an HTTP server?

:trollface:

**Comment by xla** [comment]
> That was actually the name of it for a good 15 minutes. Keeping it a bit more general for now as I may include SSH-related stuff in there too.

`gitd` then?

**Pull request merged**